atomic_reactor/util.py

```diff
@@ -1112,3 +1116,62 @@ def split_module_spec(module):
         'Module specification should be NAME:STREAM or NAME:STREAM:VERSION. ' +
         '(NAME-STREAM and NAME-STREAM-VERSION supported for compatibility.)'
     )
+
+
+class OSBSLogs(object):
+    def __init__(self, log):
+        self.log = log
+
+    def get_log_metadata(self, path, filename):
+        """
+        Describe a file by its metadata.
+
+        :return: dict
+        """
+
+        checksums = get_checksums(path, ['md5'])
+        metadata = {'filename': filename,
+                    'filesize': os.path.getsize(path),
+                    'checksum': checksums['md5sum'],
+                    'checksum_type': 'md5'}
+
+        return metadata
+
+    def get_log_files(self, osbs, build_id):
+        """
+        Build list of log files
+
+        :return: list, of log files
+        """
+
+        logs = None
+        output = []
+
+        # Collect logs from server
+        try:
+            logs = osbs.get_orchestrator_build_logs(build_id)
```

Comment:
This will download logs once for `koji_import` and again for `sendmail`. I think that's fine. The main benefit of downloading again is that by the time `sendmail` runs there's actually more logs available; and the error may have happened between `koji_import` and `sendmail`, e.g. `koji_tag` fails due to changes in koji.

Reply:
Okay.